### PR TITLE
3.2/develop - Feature Request #3847

### PR DESCRIPTION
**Pull request by atannus**
My First Contrib

I added a frame() method to the Image module.

Feature Request #3847

### CODE REVIEW

File: classes/kohana/image/gd.php [hunk 1]
```diff
@@ -221,6 +221,44 @@ class Kohana_Image_GD extends Image {
...
+		if ($bg_color[0] === '#')
+		{
+			// Remove the pound
+			$bg_color = substr($bg_color, 1);
+		}
+
+		if (strlen($bg_color) === 3)
+		{
+			// Convert shorthand into longhand hex notation
+			$bg_color = preg_replace('/./', '$0$0', $bg_color);
```
Comment (dyron) [inline classes/kohana/image/gd.php:239]: Why an expensive preg_replace call instead of concatenation ($bg_color.$bg_color)?

Comment (atannus) [inline classes/kohana/image/gd.php:239]: Do you know how hex color notation works?

Comment (atannus) [inline classes/kohana/image/gd.php:239]: You're proposing f00 (red) becomes f00f00. They're not the same.

Comment (dyron) [inline classes/kohana/image/gd.php:239]: Yep, didn't think of that. I'll do a condebench test for preg_replace() vs $bg_color[0].$bg_color[0].$bg_color[1].$bg_color[1].$bg_color[2].$bg_color[2].

**Comment by davecarlson** [comment]
What happened to this pull ? Did it not get added in ?

**Comment by atannus** [comment]
FYI, I've been using this patch for over a year now and it runs pretty good.
perhaps you can help with getting it in!

Cheers

On Tue, May 29, 2012 at 6:08 AM, Dave Carlson <
reply@reply.github.com

> wrote:
> 
> What happened to this pull ? Did it not get added in ?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/kohana/image/pull/7#issuecomment-5978716

## 

A ciência consiste em perturbar um sistema e analisar sua reação. Eu, sou
uma perturbação.

André Tannús | Ideas at Epungo
atannus@epungo.com.br | +55 11 8053-7636 | +55 11 2389-4360

"Somewhere, something incredible is waiting to be known."
Carl Sagan

**Comment by davecarlson** [comment]
i think u just need to reopen it and re-request it.

**Pull request closed without merging**